pymeasure/instruments/hcp/tc038.py

```diff
@@ -0,0 +1,135 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument
+from pyvisa.constants import Parity
+
+
+def _data_to_temp(data):
+    """Convert the returned hex value "data" to a temperature in °C."""
+    return int(data[7:11], 16) / 10.
+    # get the hex number, convert to int and shift the decimal sign
+
+
+registers = {'temperature': "D0002",
+             'setpoint': "D0120",
+             }
+
+
+class TC038(Instrument):
+    """
+    Communication with the HCP TC038 oven.
+
+    This is the older version with an AC power supply and AC heater.
+
+    It has parity or framing errors from time to time. Handle them in your
+    application.
+    """
+
+    def __init__(self, resourceName, address=1, timeout=1000,
+                 includeSCPI=False):
+        """
+        Initialize the communication.
+
+        Parameters
+        ----------
+        resourceName : str
+            name COM-Port.
+        address : int
+            address of the device. Should be between 1 and 99.
+        timeout : int
+            Timeout in ms.
+        """
+        super().__init__(resourceName, "TC038", timeout=timeout,
+                         write_termination="\r", read_termination="\r",
+                         parity=Parity.even,)
+        self.address = address
+
+        self.set_monitored_quantity()  # start to monitor the temperature
+
+    def _adjust_command(self, command):
```

Comment:
Great that you extract a method to avoid repetition!
This does not use `self`, so it should be better drop that and decorate it with `@staticmethod`.
```suggestion
    @staticmethod
    def _adjust_command(command):
```

Reply:
It does use `self` in the format string. That is exactly the reason, why I had to do this extra work.

Reply:
Ha, shows I haven't had enough coffee, yet. :-/